### DISCUSSION TITLE
Prism is now Dual License!!!

### DISCUSSION THREAD

**Discussion by brianlagunas** [PrismLibrary/Prism#2964] [General]
Prism has officially launched as a dual license product. This means that there are two types of licenses now; Community and Commercial.

Companies and individuals with less than $1 million USD in annual gross revenue, or have never received more than $3 million USD in capital from an outside source, such as private equity or venture capital, are eligible for the Community License.

Prism sells two types of commercial licenses:
- Commercial - $299 USD per developer per year
- Commercial Plus - $499 USD per developer per year.

For more information on what each license provides, please [visit our website](https://prismlibrary.com/)

Please be patient as we enter this new phase of the Prism Library. We are still working behind the scenes to not only improve the framework, but to build an infrastructure to support our commercial license holders.

If you are interested in purchasing a commercial license, but would like more information, please email us at support@prismlibrary.com.

**Comment by hafzaal** [comment]
What about forks of Prism for other frameworks that aren't officially supported e.g. the Avalonia UI fork? Is that allowed to function as normal or would they be going against the license by releasing a Prism v9 fork for Avalonia?

**Reply by brianlagunas** [reply]
They would be going against the Prism license. However, we can work with the Avalonia team to create a version of Prism for the Community License.

**Reply by hafzaal** [reply]
That makes sense, thanks for the clarification. Although the Avalonia fork for prism has been a community effort so far so I'm unsure how willing Avalonia is to spend resources towards this. Would be great if we can get something official though considering Avalonia is one of the best ways to develop multi-platform desktop apps these days.

**Reply by brianlagunas** [reply]
This is one of the goals of moving to a paid commercial license. By having companies pay for their commercial use, we hope to have the funds to hire people to add support for more platforms like Avaolnia, more tooling, improved docs and samples, etc.

**Reply by hafzaal** [reply]
I completely understand. Prism is really amazing and its been great working with it. It makes perfect sense to have this license to help support the development of the project going forward. Just unfortunate for the project I'm working on as I'm assuming the Avalonia fork will cease development soon. I guess I better find an alternative to do everything Prism does, which is going to be a challenge, haha.

**Reply by brianlagunas** [reply]
We are still trying to figure things out, and we are open to make exceptions. We could possibly have a temporary agreement for Avalonia that will only apply for community license holders. It's definitely something we want to work with the community to figure out.

**Reply by pasha-o** [reply]
hi @brianlagunas 
1- is there any comparison list of features available to see what's missing from the community edition versus other tiers?
2- is there a release nugget for the community?

**Reply by brianlagunas** [reply]
You can visit the website to see what's missing https://prismlibrary.com/#pricing

There will be a community release soon, but in the future, license holders get more request releases, and access to CI builds. This means the community will get fewer releases throughout the year.

**Comment by bennoremec** [comment]
I find the per seat license not very convenient.  A site license would be much more convenient for us.

**Reply by brianlagunas** [reply]
We'd be more than happy to discuss a site license. You can email us at support@prismlibrary.com

**Comment by FirehawkV21** [comment]
I'm currently using Prism on a few projects. They aren't commercial projects (one of them is open source). Do I need a license before I continue using it?

**Reply by brianlagunas** [reply]
Do you make more than $1,000,000 USD a year?

**Reply by FirehawkV21** [reply]
No.

**Reply by brianlagunas** [reply]
That's too bad. I wish you did 😄 

You qualify for the Community License. So you still get to use it for free.

Good luck on making your first $1M 👍

**Comment by MakeGywer** [comment]
I’m currently using Prism release v8.1.97, like most of us, I believe. This release, along with all the versions up until the last .NET Foundation release, was under the MIT License. Could you please confirm if the MIT License still applies to these older releases?

Please don’t misunderstand me, I fully support the idea that large corporations should pay for commercial usage. In fact, I’m using Prism in my company as well, and although our product is still in development, I’m confident that we will purchase licenses in the future.

My main concern at this moment is to understand whether the MIT License still applies to the usage of releases up until the last .NET Foundation release or if there’s an immediate need for us to adjust to the new licensing terms.

**Reply by brianlagunas** [reply]
You are correct. Technically the last DNF release was under MIT. Only the latest version since the license change will be impacted by the new license. So technically, you can stay on the old version and never upgrade and you'll be good licensing wise. v9 and above will require the new licensing agreement.

**Comment by mrwhy-orig** [comment]
I'd like to use Prism in my ongoing projects. I (unfortunately) am far away from the 1 million $ gross revenue, but some of my customers aren't. Do I have to buy the commercial license to build for customers or sell to customers with higher gross revenue?
Thanks for this awesome library anyway ;)

**Reply by brianlagunas** [reply]
Your customers will be required to purchase a license. They can buy it themselves directly from the Prism Library website, or you can buy it and sell it to them with a markup and act as a "reseller". Which ever you prefer.

**Reply by mrwhy-orig** [reply]
Just to clarify, if I want to sell a product or build for a customer, which uses this library with v9 ongoing, I would have to charge my customer a yearly fee, so he can use this software? Is that right? It just feels a bit strange.

**Reply by brianlagunas** [reply]
If your customer owns the code, they need to purchase a license which will be assigned to you so you can work on their code. If you own the code and sell it, and you make more than $1M then only you need a license. Otherwise, you fall within the Community license. Does that help?

**Reply by mrwhy-orig** [reply]
Ahh, ok, thanks for this clarification!

**Comment by segor** [comment]
We are going to buy commercial licenses and need to undestand how many seats we need. Do we need to buy a  license for the every developer who compiles and debug the whole c# solution  or just for people working on UI c# projects? Do we need to buy separate licenses for build automation servers? If yes, what kind of license is needed?

I tried to find the answer in the license pdf file but it is not available- 404 error.

Thanks

**Reply by brianlagunas** [reply]
Great question. We are working on an FAQ for our website, and this would be a good one to add to it.

You will need a license for each developer that debugs or compiles the code. You do not need to buy separate licenses for the build automation servers. As long as you have proper license coverage for your devs, the build servers are covered.

We are in the middle of moving things around, but here is the link to the license agreement. This will change in the future, but it's a temporary link until we get our CDN working properly.

https://cdn.prismlibrary.com/downloads/prism_license.pdf

Let me know if you have any other questions

**Reply by segor** [reply]
Thanks for answer. It is clear now.

**Reply by rczubala-BPS** [reply]
@brianlagunas would you attach an updated PDF. The link is no longer valid.

**Reply by dansiegel** [reply]
You can find the link on the bottom of the Prism website.

**Comment by thomasgalliker** [comment]
Dear @dansiegel and @brianlagunas 
thanks for the great effort you invest daily in maintaining the Prism library! I used the library in the past with WPF and Xamarin Forms for educational purposes. I'm teaching desktop and mobile app delevopment at the [University of Applied Sciences of Eastern Switzerland](https://www.ost.ch/de/weiterbildung/weiterbildungsangebot/informatik/das-net-software-engineering/modul-mobile-applications). A rather small but important part is the use of app frameworks such as Prism. Do you have a special licensing for educational purposes or would you think of adding such a license? I would kindly appreciate. Thanks for your response.

**Reply by dansiegel** [reply]
I would tend to think the real question would be, what exactly are you looking for? 

In general terms educational purposes (i.e. teaching app development in a school setting for non-commercial purposes) everyone should qualify for the Community License.

If the concern is more about accessing the private NuGet packages then I would say please contact us at support@prismlibrary.com and we can discuss your specific situation.

**Comment by cgohier** [comment]
Hi @brianlagunas  and @dansiegel,

is the commercial license needed for non-profit organizations?

Thanks a lot for your efforts.

**Reply by brianlagunas** [reply]
Yes, if the non-profit does not qualify for the community license, then they need to purchase a commercial license.

**Comment by Coolkeke** [comment]
Hello!
I am a developer from xx company and currently want to use Prism in my project. The purpose of this project is to provide a simple auxiliary tool for internal use, without any involvement in commercial development or profit. However, I have successfully applied for a community license. Can I use Prism to develop projects normally

**Reply by brianlagunas** [reply]
If xx company does not qualify for the community license, they must purchase a commercial license regardless of if the apps are internal auxiliary apps.
